isis/tests/OsirisRexOCamsCameraTests.cpp

```diff
@@ -42,34 +42,57 @@ TEST_F(OsirisRexCube, PolyMath) {
   EXPECT_NEAR(cam->UniversalLatitude(), 9.26486, 0.0001);
   EXPECT_NEAR(cam->UniversalLongitude(), 276.167, 0.0001);
 
-  EXPECT_TRUE(cam->SetUniversalGround(cam->UniversalLatitude(), cam->UniversalLongitude())); 
+  EXPECT_TRUE(cam->SetUniversalGround(cam->UniversalLatitude(), cam->UniversalLongitude()));
   EXPECT_NEAR(cam->Line(), 5, 0.01);
   EXPECT_NEAR(cam->Sample(), 5, 0.01);
 }
 
 
 TEST_F(OsirisRexCube, MappingCam) {
-  setInstrument("-64361", "PolyCam");
+  setInstrument("-64361", "MapCam");
 
-  OsirisRexOcamsCamera *cam = (OsirisRexOcamsCamera *)testCube->camera();;
-  EXPECT_PRED_FORMAT2(AssertQStringsEqual, cam->spacecraftNameLong(), "OSIRIS-REx"); 
+  OsirisRexOcamsCamera *cam = (OsirisRexOcamsCamera *)testCube->camera();
+  EXPECT_PRED_FORMAT2(AssertQStringsEqual, cam->spacecraftNameLong(), "OSIRIS-REx");
   EXPECT_PRED_FORMAT2(AssertQStringsEqual, cam->spacecraftNameShort(), "OSIRIS-REx");
   EXPECT_PRED_FORMAT2(AssertQStringsEqual, cam->instrumentNameLong(), "Mapping Camera");
   EXPECT_PRED_FORMAT2(AssertQStringsEqual, cam->instrumentNameShort(), "MapCam");
-  EXPECT_PRED_FORMAT2(AssertQStringsEqual, cam->instrumentId(), "PolyCam");
+  EXPECT_PRED_FORMAT2(AssertQStringsEqual, cam->instrumentId(), "MapCam");
 }
 
 
 TEST_F(OsirisRexCube, SamplingCam) {
-  setInstrument("-64362", "PolyCam");
+  setInstrument("-64362", "SamCam");
+
+  OsirisRexOcamsCamera *cam = (OsirisRexOcamsCamera *)testCube->camera();
+
+  EXPECT_EQ(cam->instrumentRotation()->Frame(), -27002);
 
-  OsirisRexOcamsCamera *cam = (OsirisRexOcamsCamera *)testCube->camera();;
-  EXPECT_PRED_FORMAT2(AssertQStringsEqual, cam->spacecraftNameLong(), "OSIRIS-REx"); 
+  // Test kernel IDs
+  EXPECT_EQ(cam->CkFrameId(), -64000);
+  EXPECT_EQ(cam->CkReferenceId(), 1);
+  EXPECT_EQ(cam->SpkTargetId(), -64);
+  EXPECT_EQ(cam->SpkReferenceId(), 1);
+
+  EXPECT_PRED_FORMAT2(AssertQStringsEqual, cam->spacecraftNameLong(), "OSIRIS-REx");
   EXPECT_PRED_FORMAT2(AssertQStringsEqual, cam->spacecraftNameShort(), "OSIRIS-REx");
   EXPECT_PRED_FORMAT2(AssertQStringsEqual, cam->instrumentNameLong(), "Sampling Camera");
   EXPECT_PRED_FORMAT2(AssertQStringsEqual, cam->instrumentNameShort(), "SamCam");
-  EXPECT_PRED_FORMAT2(AssertQStringsEqual, cam->instrumentId(), "PolyCam");
-   
+  EXPECT_PRED_FORMAT2(AssertQStringsEqual, cam->instrumentId(), "SamCam");
 
-}
+  const PvlGroup &inst = testCube->label()->findGroup("Instrument", Pvl::Traverse);
+  double exposureDuration = ((double) inst["ExposureDuration"])/1000;
+  QString stime = inst["StartTime"];
+  double et; // StartTime keyword is the center exposure time
+  str2et_c(stime.toLatin1().data(), &et);
```

Comment:
I think we are trying to get away from using  spice calls if possible. You might want to give this a try: 
```
cam->SetImage (0.5, 0.5); // for start time
cam->time().Et() // returns the current time
```

Reply:
@kaitlyndlee Is the spice call you are referring to `str2et_c`? I am not sure how to exactly implement this. My understanding is that we are testing the StartTime label number here. So that is why we are converting the string to et with `str2et_c` function. I am not sure on this since I am basing this test off of the Polycam unit test and all of the cam unit tests, which are also using `str2et_c`.

I did try `double et = cam->time().Et();` and it returned different outputs for the shutter time tests. If this is expected and the way you are suggesting to implement this, I can go ahead with these changes.

Reply:
@kaitlyndlee's solution sounds right. I figure there is a slight discrepancy between the clock counts and the label start time in the instrument group.